<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
    <class>MainWindow</class>
    <widget class="QMainWindow" name="MainWindow">
        <property name="geometry">
            <rect>
                <x>0</x>
                <y>0</y>
                <width>1200</width>
                <height>800</height>
            </rect>
        </property>
        <property name="windowTitle">
            <string>Trading Terminal</string>
        </property>
        <widget class="QWidget" name="centralwidget">
            <layout class="QVBoxLayout" name="mainVerticalLayout">
                <item>
                    <layout class="QHBoxLayout" name="tabControlLayout">
                        <item>
                            <widget class="QTabWidget" name="symbolTabs">
                                <property name="tabsClosable">
                                    <bool>true</bool>
                                </property>
                                <property name="movable">
                                    <bool>true</bool>
                                </property>
                                <widget class="QWidget" name="btcTab">
                                    <attribute name="title">
                                        <string>BTC/USD</string>
                                    </attribute>
                                    <layout class="QVBoxLayout" name="verticalLayout_2">
                                        <item>
                                            <widget class="QSplitter" name="orderBookSplitter">
                                                <property name="orientation">
                                                    <enum>Qt::Horizontal</enum>
                                                </property>
                                                <widget class="QTableWidget" name="bidsTable">
                                                    <property name="styleSheet">
                                                        <string notr="true">QTableWidget { color: #2ecc71; }</string>
                                                    </property>
                                                    <column>
                                                        <property name="text">
                                                            <string>Price</string>
                                                        </property>
                                                    </column>
                                                    <column>
                                                        <property name="text">
                                                            <string>Size</string>
                                                        </property>
                                                    </column>
                                                </widget>
                                                <widget class="QTableWidget" name="asksTable">
                                                    <property name="styleSheet">
                                                        <string notr="true">QTableWidget { color: #e74c3c; }</string>
                                                    </property>
                                                    <column>
                                                        <property name="text">
                                                            <string>Price</string>
                                                        </property>
                                                    </column>
                                                    <column>
                                                        <property name="text">
                                                            <string>Size</string>
                                                        </property>
                                                    </column>
                                                </widget>
                                            </widget>
                                        </item>
                                        <item>
                                            <layout class="QHBoxLayout" name="summaryLayout">
                                                <item>
                                                    <widget class="QLabel" name="lastPriceLabel">
                                                        <property name="text">
                                                            <string>Last Price: N/A</string>
                                                        </property>
                                                    </widget>
                                                </item>
                                                <item>
                                                    <widget class="QLabel" name="bboLabel">
                                                        <property name="text">
                                                            <string>BBO: N/A / N/A</string>
                                                        </property>
                                                        <property name="alignment">
                                                            <set>Qt::AlignHCenter|Qt::AlignVCenter</set>
                                                        </property>
                                                    </widget>
                                                </item>
                                                <item>
                                                    <widget class="QLabel" name="spreadLabel">
                                                        <property name="text">
                                                            <string>Spread: N/A</string>
                                                        </property>
                                                        <property name="alignment">
                                                            <set>Qt::AlignRight|Qt::AlignVCenter</set>
                                                        </property>
                                                    </widget>
                                                </item>
                                            </layout>
                                        </item>
                                        <item>
                                            <layout class="QHBoxLayout" name="orderEntryLayout">
                                                <item>
                                                    <widget class="QLabel" name="priceLabel">
                                                        <property name="text">
                                                            <string>Price:</string>
                                                        </property>
                                                    </widget>
                                                </item>
                                                <item>
                                                    <widget class="QSpinBox" name="priceSpinBox">
                                                        <property name="maximum">
                                                            <number>10000000</number>
                                                        </property>
                                                    </widget>
                                                </item>
                                                <item>
                                                    <widget class="QLabel" name="qtyLabel">
                                                        <property name="text">
                                                            <string>Quantity:</string>
                                                        </property>
                                                    </widget>
                                                </item>
                                                <item>
                                                    <widget class="QSpinBox" name="qtySpinBox">
                                                        <property name="maximum">
                                                            <number>10000000</number>
                                                        </property>
                                                    </widget>
                                                </item>
                                                <item>
                                                    <widget class="QPushButton" name="buyButton">
                                                        <property name="text">
                                                            <string>BUY</string>
                                                        </property>
                                                        <property name="styleSheet">
                                                            <string notr="true">background-color: #2ecc71; color: white;
                                                                font-weight: bold; padding: 8px;
                                                            </string>
                                                        </property>
                                                    </widget>
                                                </item>
                                                <item>
                                                    <widget class="QPushButton" name="sellButton">
                                                        <property name="text">
                                                            <string>SELL</string>
                                                        </property>
                                                        <property name="styleSheet">
                                                            <string notr="true">background-color: #e74c3c; color: white;
                                                                font-weight: bold; padding: 8px;
                                                            </string>
                                                        </property>
                                                    </widget>
                                                </item>
                                            </layout>
                                        </item>
                                        <item>
                                            <widget class="QTabWidget" name="userDataTabs">
                                                <widget class="QWidget" name="ordersTab">
                                                    <attribute name="title">
                                                        <string>My Open Orders</string>
                                                    </attribute>
                                                    <layout class="QVBoxLayout" name="ordersLayout">
                                                        <item>
                                                            <widget class="QTableWidget" name="ordersTable">
                                                                <column>
                                                                    <property name="text">
                                                                        <string>Timestamp</string>
                                                                    </property>
                                                                </column>
                                                                <column>
                                                                    <property name="text">
                                                                        <string>Order ID</string>
                                                                    </property>
                                                                </column>
                                                                <column>
                                                                    <property name="text">
                                                                        <string>Side</string>
                                                                    </property>
                                                                </column>
                                                                <column>
                                                                    <property name="text">
                                                                        <string>Price</string>
                                                                    </property>
                                                                </column>
                                                                <column>
                                                                    <property name="text">
                                                                        <string>Quantity</string>
                                                                    </property>
                                                                </column>
                                                                <column>
                                                                    <property name="text">
                                                                        <string>Status</string>
                                                                    </property>
                                                                </column>
                                                            </widget>
                                                        </item>
                                                        <item>
                                                            <layout class="QHBoxLayout" name="orderActionsLayout">
                                                                <item>
                                                                    <spacer name="horizontalSpacer">
                                                                        <property name="orientation">
                                                                            <enum>Qt::Horizontal</enum>
                                                                        </property>
                                                                        <property name="sizeHint" stdset="0">
                                                                            <size>
                                                                                <width>40</width>
                                                                                <height>20</height>
                                                                            </size>
                                                                        </property>
                                                                    </spacer>
                                                                </item>
                                                                <item>
                                                                    <widget class="QPushButton"
                                                                            name="modifyOrderButton">
                                                                        <property name="toolTip">
                                                                            <string>Modify the price/quantity of the
                                                                                selected order
                                                                            </string>
                                                                        </property>
                                                                        <property name="styleSheet">
                                                                            <string notr="true">background-color:
                                                                                #f39c12; color: white; font-weight:
                                                                                bold; padding: 5px;
                                                                            </string>
                                                                        </property>
                                                                        <property name="text">
                                                                            <string>Modify Selected Order</string>
                                                                        </property>
                                                                    </widget>
                                                                </item>
                                                                <item>
                                                                    <widget class="QPushButton"
                                                                            name="cancelOrderButton">
                                                                        <property name="toolTip">
                                                                            <string>Cancel the selected order</string>
                                                                        </property>
                                                                        <property name="styleSheet">
                                                                            <string notr="true">background-color:
                                                                                #e74c3c; color: white; font-weight:
                                                                                bold; padding: 5px;
                                                                            </string>
                                                                        </property>
                                                                        <property name="text">
                                                                            <string>Cancel Selected Order</string>
                                                                        </property>
                                                                    </widget>
                                                                </item>
                                                            </layout>
                                                        </item>
                                                    </layout>
                                                </widget>
                                                <widget class="QWidget" name="tradesTab">
                                                    <attribute name="title">
                                                        <string>Trade History</string>
                                                    </attribute>
                                                    <layout class="QVBoxLayout" name="tradesLayout">
                                                        <item>
                                                            <widget class="QTableWidget" name="tradesTable">
                                                                <column>
                                                                    <property name="text">
                                                                        <string>Timestamp</string>
                                                                    </property>
                                                                </column>
                                                                <column>
                                                                    <property name="text">
                                                                        <string>Trade ID</string>
                                                                    </property>
                                                                </column>
                                                                <column>
                                                                    <property name="text">
                                                                        <string>Price</string>
                                                                    </property>
                                                                </column>
                                                                <column>
                                                                    <property name="text">
                                                                        <string>Quantity</string>
                                                                    </property>
                                                                </column>
                                                                <column>
                                                                    <property name="text">
                                                                        <string>Source Order ID</string>
                                                                    </property>
                                                                </column>
                                                                <column>
                                                                    <property name="text">
                                                                        <string>Matched Order ID</string>
                                                                    </property>
                                                                </column>
                                                            </widget>
                                                        </item>
                                                    </layout>
                                                </widget>
                                                <widget class="QWidget" name="logTab">
                                                    <attribute name="title">
                                                        <string>Logs</string>
                                                    </attribute>
                                                    <layout class="QVBoxLayout" name="logLayout">
                                                        <item>
                                                            <widget class="QTextEdit" name="logText">
                                                                <property name="readOnly">
                                                                    <bool>true</bool>
                                                                </property>
                                                            </widget>
                                                        </item>
                                                    </layout>
                                                </widget>
                                            </widget>
                                        </item>
                                    </layout>
                                </widget>
                            </widget>
                        </item>
                        <item>
                            <widget class="QPushButton" name="addTabButton">
                                <property name="text">
                                    <string>+</string>
                                </property>
                                <property name="toolTip">
                                    <string>Add New Symbol Tab</string>
                                </property>
                                <property name="maximumSize">
                                    <size>
                                        <width>30</width>
                                        <height>30</height>
                                    </size>
                                </property>
                                <property name="minimumSize">
                                    <size>
                                        <width>30</width>
                                        <height>30</height>
                                    </size>
                                </property>
                                <property name="styleSheet">
                                    <string notr="true">
                                        QPushButton {
                                        background-color: #3498db;
                                        color: white;
                                        border: none;
                                        border-radius: 15px;
                                        font-weight: bold;
                                        font-size: 14px;
                                        }
                                        QPushButton:hover {
                                        background-color: #2980b9;
                                        }
                                        QPushButton:pressed {
                                        background-color: #21618c;
                                        }
                                    </string>
                                </property>
                            </widget>
                        </item>
                    </layout>
                </item>
            </layout>
        </widget>
        <widget class="QStatusBar" name="statusbar"/>
        <widget class="QToolBar" name="toolBar">
            <property name="windowTitle">
                <string>toolBar</string>
            </property>
            <attribute name="toolBarArea">
                <enum>TopToolBarArea</enum>
            </attribute>
            <addaction name="actionConnect"/>
            <addaction name="actionDisconnect"/>
            <addaction name="actionAddSymbol"/>
        </widget>
        <action name="actionConnect">
            <property name="text">
                <string>Connect</string>
            </property>
        </action>
        <action name="actionDisconnect">
            <property name="text">
                <string>Disconnect</string>
            </property>
        </action>
        <action name="actionAddSymbol">
            <property name="text">
                <string>Add Symbol</string>
            </property>
            <property name="shortcut">
                <string>Ctrl+T</string>
            </property>
        </action>
    </widget>
    <resources/>
    <connections/>
</ui>